### PR TITLE
Database week2

### PR DESCRIPTION
**Pull request by shobanamg**

### CODE REVIEW

File: databases/week2/part1.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+where name like 'Anamika%';
+
+select *
+from status;
+
+select *
+from task;
+
+select * from user_task where user_id = 12;
+select * from user_task where task_id = 1;
```
Comment (Elia-Git) [inline databases/week2/part1.sql]: This testing part can be removed as they are not  part of the hw

File: databases/week2/part1.sql [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+/*  Add a task with these attributes: title, description, created, updated, due_date, status_id, user_id */
+insert into task (title, description, created, updated, due_date, status_id)
+values ('read a book', 'Maybe something about parenting?', '2023-01-11 06:14:01', '2023-01-11 06:14',
+        '2023-02-11 06:14:01', 2);
```
Comment (Elia-Git) [inline databases/week2/part1.sql]: You forgot to insert user_id here as per the Q. Better to test before post
```
insert into task (title, description, created, updated, due_date, status_id, user_id)
values ('read a book', 'Maybe something about parenting?', '2023-01-11 06:14:01', '2023-01-11 06:14', '2023-02-11 06:14:01', 1, 2);
```

File: databases/week2/part1.sql [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+insert into user (name, email, phone)
+values ('Anamika Joseph', 'anamikajoseph0@harvard.edu', '123-456-7890');
+
+insert into user_task (user_id, task_id)
+values (12, 36);
```
Comment (Elia-Git) [inline databases/week2/part1.sql]: No clue why you need this..? Do you want to create `user_task `& `user` tables..? They are already created.
all attributes: `title, description, created, updated, due_date, status_id, user_id` can be inserted into table task

File: databases/week2/part1.sql [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+/* Change the title of a task */
+UPDATE task
+SET title = 'do exercise'
+WHERE id = 36;
+
+/* Change a task due date */
+UPDATE task
+SET due_date = '2023-12-22 20:58:03'
+WHERE status_id <> 3;
```
Comment (Elia-Git) [inline databases/week2/part1.sql:26]: Here you are updating due_date for all tasks having `status_id=3` , is this intentionally? 
Easier & better to update a task with it id
let say
```
UPDATE task
SET due_date = '2023-12-22 20:58:03'
WHERE id <> 36;
``` in this case

File: databases/week2/part1.sql [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+WHERE id = 5;
+
+/* Mark a task as complete */
+UPDATE task
+SET status_id = 3
+WHERE id = 5;
+
+/* Delete a task */
+delete from task
+where id = 36;
```
Comment (Elia-Git) [inline databases/week2/part1.sql]: good here!

File: databases/week2/part2.sql [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+insert into student ( name, email, phone, class_id) values ( 'Hosea Jack', 'hoseajack@nymag.com',  '302-678-7931' , 2);
+insert into student ( name, email, phone, class_id) values ( 'Jeffie Pernell', 'jfrenold@comcast.net', '891-952-6749' ,3);
+
+
+/* advanced tasks: */
+CREATE INDEX index_name
+    ON student (name);
+
+ALTER TABLE class
+    ADD status ENUM('not-started', 'ongoing', 'finished') not null default 'not-started';
```
Comment (Elia-Git) [inline databases/week2/part2.sql:54]: Good job here!

File: databases/week2/part3.sql [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+insert into student ( name, email, phone, class_id) values ( 'Hosea Jack', 'hoseajack@nymag.com',  '302-678-7931' , 2);
+insert into student ( name, email, phone, class_id) values ( 'Jeffie Pernell', 'jfrenold@comcast.net', '891-952-6749' ,3);
+
+
+/* advanced tasks: */
+CREATE INDEX index_name
+    ON student (name);
+
+ALTER TABLE class
+    ADD status ENUM('not-started', 'ongoing', 'finished') not null default 'not-started';
```
Comment (Elia-Git) [inline databases/week2/part3.sql]: This part (part 3 ) is a copy of part . Remove one of them

File: databases/week2/food_villa.sql [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+USE food_villa;
+
+CREATE TABLE customer (
+                          customer_id INT(11) AUTO_INCREMENT PRIMARY KEY,
+                          first_name VARCHAR(30) NOT NULL,
+                          last_name VARCHAR(30) NOT NULL,
+                          middle_name VARCHAR(30) NOT NULL,
+                          email VARCHAR(50) NOT NULL,
+                          phone_number VARCHAR(15) NOT NULL,
```
Comment (Elia-Git) [inline databases/week2/food_villa.sql:14]: Line 14 , good to use varchar/string datatype for phone_number, but is good to know why as we often use int for number ?

**Review by Elia-Git — CHANGES_REQUESTED** [review]
Good job in general. Cool you use indexing in the self built database too. Please consider the given comments. Thx.

File: databases/week2/part1.sql (hunk 2 above)
Comment (shobanamg) [reply databases/week2/part1.sql]: I have auto increment set to the id column for the task table, that's why Id is not set manually

File: databases/week2/part1.sql (hunk 4 above)
Comment (shobanamg) [reply databases/week2/part1.sql:26]: I was updating due_date for all the tasks which were not done(status=3) to a future date. Basically i was trying to update the due_date for all the tasks which were in 'Not started' and 'In progress' .

File: databases/week2/part1.sql (hunk 1 above)
Comment (shobanamg) [reply databases/week2/part1.sql]: I don't know if you have reviewed each commit. If you see the latest files, this has been already deleted

File: databases/week2/part1.sql (hunk 3 above)
Comment (shobanamg) [reply databases/week2/part1.sql]: Initially i have done this part with the hyfmodel2 database, then refactored it with hyfmodels database. So if you see the latest files it won't there.

File: databases/week2/part3.sql (hunk 7 above)
Comment (shobanamg) [reply databases/week2/part3.sql]: I don't understand, can you elaborate?

File: databases/week2/part1.sql (hunk 2 above)
Comment (shobanamg) [reply databases/week2/part1.sql]: create table task
(
    id          int unsigned auto_increment
        primary key,
    title       varchar(255) not null,
    description text         null,
    created     datetime     not null,
    updated     datetime     not null,
    due_date    datetime     null,
    status_id   int unsigned not null,
    user_id     int unsigned null,
    constraint fk_status
        foreign key (status_id) references status (id)
            on delete cascade,
    constraint fk_user
        foreign key (user_id) references user (id)
            on delete cascade
)
    collate = utf8mb4_unicode_ci;

**Pull request merged**